### PR TITLE
build: delegate javascript workspace build to nx

### PR DESCRIPTION
**Pull request by mswertz**
motivation:
* reduce configuration needed to add new apps (i.e. amount of gradle needed)
* automatic build dependencies between apps (i.e. when adding libs)
* javascript so probably easier for javascript 

change:
* using https://nx.dev/recipes/adopting-nx/adding-to-monorepo

What do you think? is this better then doing the workspace in java?

### CODE REVIEW

**Comment by mswertz** [comment]
Before I merge I would like to double check that the dev experience with nuxt works allright.

**Review by connoratrug — CHANGES_REQUESTED** [review]
Most of the stuff seams to work, i checked building via gradle , building via yarn , running via Intelij. 

One issue i got was the logo not being rendered 
Screenshot 2023-08-31 at 13 24 49

Need to check if this is only a issue on my machine.

I could not really check if the CI (ssr-catalogue)image building work, looks like it works in the preview. But i suggest we pick a good time to merge and then check the images that get build after the merge.

**Comment by mswertz** [comment]
logo works. This has no impact on existing projects in its current form.
Next I would like to reorganize in a frontend/libs and frontend/apps structure so we dont need to change build.gradle for new projects

File: azure-pipelines.yml [hunk 1]
```diff
@@ -36,7 +36,7 @@ steps:
 - script: |
     export JAVA_HOME=/usr/lib/jvm/temurin-17-jdk-amd64/
     ./gradlew -version
-    ./gradlew test jacocoMergedReport checkFormat spotlessCheck 
+    ./gradlew test jacocoMergedReport
```
Comment (connoratrug) [inline azure-pipelines.yml:39]: removed ?

**Review by connoratrug — APPROVED** [review]
looks good , every things looks like its building and apps are running.

some small notes, see comments

**Comment by connoratrug** [comment]
Screenshot 2023-09-21 at 09 27 10

**Review by connoratrug — CHANGES_REQUESTED** [review]
put back ci sonar scan

File: azure-pipelines.yml (hunk 1 above)
Comment (mswertz) [reply azure-pipelines.yml:39]: only from the azure pipeline to make it faster, unncessery change for this PR anyway so will remove

**Review by connoratrug — APPROVED** [review]

**Pull request merged**